src/EnliteMonolog/Service/MonologServiceFactory.php

```diff
@@ -57,6 +58,10 @@ public function createLogger($container, MonologOptions $options)
             $logger->pushProcessor($this->createProcessor($container, $processor));
         }
 
+        if (class_exists('\Monolog\ErrorHandler') && $options->isErrorHandler()) {
+            ErrorHandler::register($logger);
```

Comment:
great point!

the module provides logger configuration which the application (or other modules) can (and should be able to) override. this means we need to wait until after the config is loaded to auto-register a logger as an error-handler, or else we might force implementation.

merged config is available via service locator in bootstrap, but i think we would need to bind to the config merge event in the init method. i could scan the enlitepro logger configurations for error loggers and fetch those names from the service manager at init/boot time so that they are registered and ready to go. as long as the merged config AND service locator are available in either init or bootstrap then it should work. i know service locator and merged config are available in module bootstrap, but i think only the merged config is available in init by binding to the merged config event.

can you offer any guidance?

aside from auto-registration at boot/init, we could provide documentation guiding devs to fetch the error-logger as early as possible in their application module.